src/fr/scanmanga/src/eu/kanade/tachiyomi/extension/fr/scanmanga/ScanManga.kt

```diff
@@ -46,23 +41,36 @@ class ScanManga : ParsedHttpSource() {
         }.build()
 
     private val json: Json by injectLazy()
+    private fun shuffle(s: String): String {
+        val chars = s.toMutableList()
+        chars.shuffle()
+        return chars.joinToString("")
+    }
 
     override fun headersBuilder(): Headers.Builder = super.headersBuilder()
         .add("Accept-Language", "fr-FR")
+        .set(
+            "User-Agent",
+            "Mozilla/5.0 (Linux; Android 16; LM-X420) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/138.0.7204.158 Mobile Safari/537.36",
+        )
 
     // Popular
     override fun popularMangaRequest(page: Int): Request {
-        return GET("$baseUrl/TOP-Manga-Webtoon-22.html", headers)
+        return GET("$baseUrl/TOP-Manga-Webtoon-36.html", headers)
     }
 
-    override fun popularMangaSelector() = "div.image_manga a[href]"
+    override fun popularMangaSelector() = "div.top"
 
     override fun popularMangaFromElement(element: Element): SManga {
-        return SManga.create().apply {
-            title = element.select("img").attr("title")
-            setUrlWithoutDomain(element.attr("href"))
-            thumbnail_url = element.select("img").attr("data-original")
-        }
+        val manga = SManga.create()
+
+        val titleElement = element.selectFirst("a.atop")
+        val imgElement = element.selectFirst("a > img")
+
+        manga.setUrlWithoutDomain(titleElement?.attr("href")?.removePrefix(baseUrl) ?: "")
+        manga.title = titleElement?.text()?.trim() ?: "Titre inconnu"
```

Comment:
This is a critical field, should not just silently ignore if unavailable
```suggestion
        manga.title = titleElement?.text()!!
```

Reply:
When I make this change, the "Popular" tab shows: `NullPointerException: null`.
Also, there's another issue I can't fix  a `null` manga gets added as the first element (you can see it in the screenshot).

Btw, thanks a lot for your feedback! :)

![Uploading Screenshot_20250803_224111.png…]()

Reply:
Then you should adjust the Jsoup selector so it doesn't pick up the invalid entry. I don't have direct access to the site, but you could probably try with:
```kt
override fun popularMangaSelector() = "div.top a[href]"
```